com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs

```diff
@@ -130,15 +130,9 @@ static HDProjectSettings instance
             }
         }
 
-        HDProjectSettings()
-        {
-            // s_Instance.FillPresentPluginMaterialVersions(); <
-            // We can't call this here as the scriptable object will not be deserialized in yet
-        }
-
         //// We force the instance to be loaded/created and ready with valid values on assembly reload.
         [InitializeOnLoadMethod]
-        static void Reset()
+        static void InitializeFillPresentPluginMaterialVersionsOnLoad()
```

Comment:
@slunity When you named it Reset in https://github.com/Unity-Technologies/Graphics/commit/99384cf466fe3074e6e712f692a9191fb8f06084 , Was it to be also called when reseting the ScriptableObject from inspector contextual menu?

Reply:
@RSlysz Good question :) I'm trying to remember if the naming was coincidental or not, although I wouldn't have put a static in front if it wasn't in the API. From what @alex-vazquez is saying, this is called by name match only, but more importantly, on load and/or create, and not only when the contextual menu is called? ie basically everytime the SO is created or loaded? In any case this change is ok, but I'm just wondering if there's a scenario where we should also hook reset to do this, I guess not, as the HDProjectSettings is not touched by the user it's not even in the project's Assets hierarchy, so no context menu interactions there.

Reply:
It was allowed to have static Reset before. Unity have changed and there is a new fence as this was causing silent issues it seams

Reply:
Ah I see. Thanks for the info!